orc8r/cloud/go/services/orchestrator/obsidian/handlers/gateway_handlers.go

```diff
@@ -11,69 +11,66 @@
  * limitations under the License.
  */
 
+// File gateway_handlers.go provides generic gateway handlers, with hooks for
+// specific gateway types.
+//
+// These handlers do not support updating a gateway's ID.
+
 package handlers
 
 import (
+	"fmt"
 	"net/http"
 
+	"github.com/go-openapi/swag"
+
 	"magma/orc8r/cloud/go/obsidian"
 	"magma/orc8r/cloud/go/orc8r"
 	"magma/orc8r/cloud/go/serde"
 	"magma/orc8r/cloud/go/services/configurator"
 	"magma/orc8r/cloud/go/services/device"
 	"magma/orc8r/cloud/go/services/orchestrator/obsidian/models"
-	"magma/orc8r/cloud/go/services/state"
+	"magma/orc8r/cloud/go/services/state/wrappers"
 	"magma/orc8r/cloud/go/storage"
 	merrors "magma/orc8r/lib/go/errors"
 
-	"github.com/go-openapi/swag"
 	"github.com/labstack/echo"
 	"github.com/pkg/errors"
 )
 
-// MagmadEncompassingGateway is an interface for a gateway API model which
-// wraps the magmad gateway with more fields that translate into additional
-// network entities in the storage layer.
-// Note that *models.MagmadGateway itself implements this interface as well.
-type MagmadEncompassingGateway interface {
-	// ValidatableModel allows the model to be validated by calling
-	// ValidateModel()
+// GatewaySubtype represents a subtype of the Magmad gateway.
+// The gateway subtype wraps the base Magmad gateway with additional fields by
+// creating and associating further network entities with the Magmad gateway.
+//
+// Note: since the default Magmad gateway model implements this interface as
+// well, DO NOT return the base Magmad model from any of the "get additional"
+// methods.
+type GatewaySubtype interface {
```

Comment:
I don't like the name change - this interface represents a composition of the Magmad gateway, it's not a subtype in an inheritance-style relationship

Reply:
Cool I'll just revert